packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormSelectFieldInput.tsx

```diff
@@ -0,0 +1,266 @@
+    });
+
+    goBackToPreviousHotkeyScope();
+  };
+
+  const [selectWrapperRef, setSelectWrapperRef] =
+    useState<HTMLDivElement | null>(null);
+
+  const [filteredOptions, setFilteredOptions] = useState<SelectOption[]>([]);
+
+  const { resetSelectedItem } = useSelectableList(
+    SINGLE_RECORD_SELECT_BASE_LIST,
+  );
+
+  const clearField = () => {
+    setDraftValue({
+      type: 'static',
+      editingMode: 'view',
+      value: '',
+    });
+  };
+
+  const selectedOption = field.metadata.options.find(
+    (option) => option.value === draftValue.value,
+  );
+
+  const handleClearField = () => {
+    clearField();
+    onCancel();
+  };
+
+  const handleSubmit = (option: SelectOption) => {
+    onSubmit(option?.value);
+
+    resetSelectedItem();
+  };
+
+  const handleUnlinkVariable = () => {
+    setDraftValue({
+      type: 'static',
+      value: '',
+      editingMode: 'view',
+    });
+
+    onPersist(null);
+  };
+
+  const handleVariableTagInsert = (variableName: string) => {
+    setDraftValue({
+      type: 'variable',
+      value: variableName,
+    });
+
+    onPersist(variableName);
+  };
+
+  const handleDisplayModeClick = () => {
+    if (draftValue.type !== 'static') {
+      throw new Error(
+        'This function can only be called when editing a static value.',
+      );
+    }
+
+    setDraftValue({
+      ...draftValue,
+      editingMode: 'edit',
+    });
+
+    setHotkeyScopeAndMemorizePreviousScope(hotkeyScope);
+  };
+
+  const handleSelectEnter = (itemId: string) => {
+    const option = filteredOptions.find((option) => option.value === itemId);
+    if (isDefined(option)) {
+      onSubmit(option.value);
+      resetSelectedItem();
+    }
+  };
+
+  useScopedHotkeys(
+    Key.Escape,
+    () => {
+      onCancel();
+      resetSelectedItem();
+    },
+    hotkeyScope,
+    [onCancel, resetSelectedItem],
+  );
+
+  const optionIds = [
+    `No ${field.label}`,
+    ...filteredOptions.map((option) => option.value),
+  ];
+
+  return (
+    <StyledFormFieldInputContainer>
+      {label ? <InputLabel>{label}</InputLabel> : null}
+
+      <StyledFormFieldInputRowContainer>
+        <StyledFormFieldInputInputContainer
+          ref={setSelectWrapperRef}
+          hasRightElement={isDefined(VariablePicker)}
+        >
+          {draftValue.type === 'static' ? (
+            <>
+              <StyledDisplayModeContainer
+                data-open={draftValue.editingMode === 'edit'}
+                onClick={handleDisplayModeClick}
+              >
+                <VisibilityHidden>Edit</VisibilityHidden>
+
+                {isDefined(selectedOption) ? (
+                  <Tag
+                    preventShrink
+                    color={selectedOption.color}
+                    text={selectedOption.label}
+                  />
+                ) : null}
+              </StyledDisplayModeContainer>
+
+              {draftValue.editingMode === 'edit' ? (
+                <SelectableList
```

Comment:
same here, can you use `<SelectFieldInput` component?

Reply:
The same reasoning applies here: https://github.com/twentyhq/twenty/pull/8815#discussion_r1867410875